src/overlays/actors/ovl_En_Zl3/z_en_zl3.c

```diff
@@ -15,421 +19,2827 @@ void EnZl3_Destroy(Actor* thisx, GlobalContext* globalCtx);
+    func_80B534CC(this);
+    func_80B56474(this, EnZl3_FrameUpdateMatrix(this));
+    func_80B564A8(this, globalCtx);
+}
+
+void func_80B56DA4(EnZl3* this) {
+    func_800788CC(NA_SE_EV_ZELDA_POWER);
+}
+
+void func_80B56DC8(EnZl3* this) {
+    func_80078914(&this->actor.projectedPos, NA_SE_VO_Z1_PAIN);
+}
+
+void func_80B56DEC(EnZl3* this) {
+    SkelAnime* skelAnime = &this->skelAnime;
+    
+    if ((skelAnime->mode == 2) && func_800A56C8(skelAnime, 9.0f) != 0) {
+        func_80078914(&this->actor.projectedPos, NA_SE_VO_Z1_OPENDOOR);
+    }
+}
+
+void func_80B56E38(EnZl3* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    s32 sfxId;
+    SkelAnime* sp20 = &this->skelAnime;
+
+    if ((func_800A56C8(sp20, 6.0f) || func_800A56C8(sp20, 0.0f)) && (this->actor.bgCheckFlags & 1)) {
+        sfxId = 0x800;
+        sfxId += func_80041F34(&globalCtx->colCtx, this->actor.floorPoly, this->actor.floorPolySource);
+        func_80078914(&this->actor.projectedPos, sfxId);
+    }
+}
+
+void func_80B56EB8(EnZl3* this, GlobalContext* globalCtx) {
+    Flags_SetSwitch(globalCtx, func_80B54DB4(this));
+}
+
+s32 func_80B56EE4(EnZl3* this, GlobalContext* globalCtx) {
+    return Flags_GetSwitch(globalCtx, func_80B54DB4(this));
+}
+
+void func_80B56F10(EnZl3* this, GlobalContext* globalCtx) {
+    s32 waypoint;
+    Path* pathHead = globalCtx->setupPathList;
+
+    if (pathHead != NULL) {
+        waypoint = func_80B54DC4(this);
+        pathHead += waypoint;
+        this->unk_30C = pathHead;
+        this->unk_310 = pathHead->count;
+        osSyncPrintf("En_Zl3_Get_path_info レールデータをゲットだぜ = %d!!!!!!!!!!!!!!\n", waypoint);
+    } else {
+        osSyncPrintf("En_Zl3_Get_path_info レールデータが無い!!!!!!!!!!!!!!!!!!!!\n");
+    }
+}
+
+s32 func_80B56F8C(EnZl3* this, s32 arg1) {
+    s32 unk_310 = this->unk_310;
+
+    if (unk_310 > arg1) {
+        return 1;
+    }
+    return 0;
+}
+
+Vec3s* func_80B56FAC(EnZl3* this, s32 arg1) {
+    Vec3s* point;
+    Path* pathList = this->unk_30C;
+
+    if ((pathList != NULL) && func_80B56F8C(this, arg1)) {
+        point = &((Vec3s*)SEGMENTED_TO_VIRTUAL(pathList->points))[arg1];
+        return point;
+    }
+    return NULL;
+}
+
+s32 func_80B57034(EnZl3* this, s32 arg1, s32 arg2) {
+    Vec3s* vec1 = func_80B56FAC(this, arg1);
+    Vec3s* vec2 = func_80B56FAC(this, arg2);
+
+    if ((vec2 != NULL) && (vec1 != NULL)) {
+        f32 xDiff = vec2->x - vec1->x;
+        f32 zDiff = vec2->z - vec1->z;
+
+        return ((xDiff == 0.0f) && (zDiff == 0.0f)) ? 0 : (s16)(Math_atan2f(xDiff, zDiff) * 10430.3779296875f);
+    }
+    return 0;
+}
+
+s16 func_80B57104(EnZl3* this, s32 arg1) {
+    Vec3s* point = func_80B56FAC(this, arg1);
+
+    if (point != NULL) {
+        f32 xDiff = point->x - this->actor.posRot.pos.x;
+        f32 zDiff = point->z - this->actor.posRot.pos.z;
+
+        if ((xDiff != 0.0f) || (zDiff != 0.0f)) {
+            return Math_atan2f(xDiff, zDiff) * 10430.3779296875f; // todo float
```

Comment:
Relevant in other places for this file
```suggestion
            return Math_atan2f(xDiff, zDiff) * (0x8000 / M_PI);
```